soaktest/soakclient/soakclient.go

```diff
@@ -0,0 +1,357 @@
+package main
+
+/**
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"os"
+	"os/signal"
+	"soaktest"
+	"sync"
+	"syscall"
+	"time"
+)
+
+const END_TO_END = "end.to.end."
+
+func main() {
+
+	broker := flag.String("broker", "127.0.0.1:9092", "bootstrap servers")
+	inputTopic := flag.String("inputTopic", "inputTopic",
+		"producer will deliver messages to this topic")
+	outTopic := flag.String("outTopic", "outTopic",
+		"consumer will consume messages from this topic")
+	groupId := flag.String("groupId", "groupId",
+		"the group consumer will join")
+	inputTopicPartitionsNum := flag.Int("inputTopicPartitionsNum", 1,
+		"inputTopic partition number")
+	outTopicPartitionsNum := flag.Int("outTopicPartitionsNum", 1,
+		"outTopic partition number")
+	replicationFactor := flag.Int("replicationFactor", 1, "topic replication")
+	ccloudAPIKey := flag.String("ccloudAPIKey", "", "sasl username")
+	ccloudAPISecret := flag.String("ccloudAPISecret", "", "sasl password")
+
+	flag.Parse()
+
+	soaktest.InitLogFiles("../log", fmt.Sprintf("../log/soakclient_%s.log",
+		time.Now().Format("01-02-2006")))
```

Comment:
Not sure what that format is (the great american DD/MM mixup), can we go with YYYY-MM-DD?